test/sanity/validate-modules/validate-modules

```diff
@@ -674,6 +674,12 @@ class ModuleValidator(Validator):
                             318,
                             'Module deprecated, but DOCUMENTATION.deprecated is missing'
                         ))
+                else:
+                    if 'module' in doc and doc['module'] != self.name:
+                        self.errors.append((
+                            319,
+                            ("'Expected 'module: %s', found 'module: %s'" % (self.name, doc['module']))
+                        ))
```

Comment:
Ideally I think I would like to see this utilize the voluptuous schema, to handle this.  Where possible using the schema would allow us to remain as consistent as possible with validating the schema of the docs.

We can do this by making `doc_schema` a factory function for building the schema.

Something like:

```
def doc_schema(module_name):
    return Schema(
        {
            Required('module'): module_name,
```

Then when we use it:

```
self._validate_docs_schema(doc, doc_schema(self.object_name), 'DOCUMENTATION', 305)
```